core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -138,15 +138,33 @@ var TopicsAndSkillsDashboardPage = function() {
     });
   };
 
-  this.createTopic = function(title) {
-    waitFor.elementToBeClickable(
-      createTopicButton,
-      'Create Topic button takes too long to be clickable');
-    createTopicButton.click();
+  this.createTopic = function(topicName, shouldCloseTopicEditor) {
+    var initialHandles = [];
+    return browser.getAllWindowHandles().then(function(handles) {
+      initialHandles = handles;
+      return browser.getWindowHandle();
+    }).then(function(parentHandle) {
+      waitFor.elementToBeClickable(
+        createTopicButton,
+        'Create Topic button takes too long to be clickable');
+      createTopicButton.click();
 
-    topicNameField.sendKeys(title);
-    confirmTopicCreationButton.click();
-    waitFor.pageToFullyLoad();
+      topicNameField.sendKeys(topicName);
+      confirmTopicCreationButton.click();
+
+      waitFor.newTabToBeCreated('Creating topic takes too long');
+      return browser.getAllWindowHandles().then(function(handles) {
+        var newHandle = handles.filter(
```

Comment:
Here, I am finding the new tab's handle by finding the difference between list of tabs initially and after topic creation and using that.